graviti/schema/builtin.py

```diff
@@ -5,20 +5,53 @@
 """The Portex builtin types."""
 
 
-from typing import Any, Iterable, List, Mapping, Optional, Sequence, Tuple, Union, overload
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    overload,
+)
+
+import pyarrow as pa
 
 import graviti.schema.ptype as PTYPE
 from graviti.schema.base import PortexType
 from graviti.schema.field import Field, Fields
 from graviti.schema.package import packages
 from graviti.schema.param import Param, Params, param
+from graviti.schema.pyarrow import BuiltinExtension
+
+if TYPE_CHECKING:
+    from pyarrow.lib import DataType as paDataType
 
 builtins = packages.builtins
 
+_TYPE_TO_PYARROW = {
+    "int": pa.int32(),
+    "float": pa.float32(),
+    "bool": pa.bool_(),
+    "str": pa.string(),
+    "int32": pa.int32(),
+    "int64": pa.int64(),
+    "float32": pa.float32(),
+    "float64": pa.float64(),
+    "boolean": pa.bool_(),
+    "string": pa.string(),
+}
+
 
 class PortexBuiltinType(PortexType):
     """The base class of Portex builtin type."""
 
+    _pyarrow_type_factory: Callable[..., "paDataType"]
```

Comment:
do we really need a factory?

Reply:
Some duplicate code to init `BuiltinExtension` can be saved.